src/test/java/de/tum/in/www1/artemis/ParticipationIntegrationTest.java

```diff
@@ -523,6 +527,113 @@ public void getParticipationForTeamExercise() throws Exception {
+    }
+
+    @Test
+    @WithMockUser(username = "student1", roles = "USER")
+    public void getParticipationByExerciseAndStudentIdForTeam() throws Exception {
+        var exercise = createTextExerciseForTeam();
+        var student = database.getUserByLogin("student1");
+        var team = createTeamForExercise(student, exercise);
+        exercise = addTeamToExercise(team, exercise);
+
+        var participation = database.addTeamParticipationForExercise(exercise, team.getId());
+        var actualParticipation = request.get("/api/exercises/" + exercise.getId() + "/participation", HttpStatus.OK, StudentParticipation.class);
+        assertThat(actualParticipation).isEqualTo(participation);
+
+        var participations = participationService.findByExerciseAndStudentId(exercise, student.getId());
+        assertThat(participations.size()).isEqualTo(1);
+        assertThat(participations.get(0).getId()).isEqualTo(participation.getId());
+    }
+
+    @Test
+    @WithMockUser(username = "student1", roles = "USER")
+    public void getParticipationByExerciseAndStudentLoginAnyStateWithEagerResultsForTeam() throws Exception {
+        var exercise = createTextExerciseForTeam();
+        var student = database.getUserByLogin("student1");
+        var team = createTeamForExercise(student, exercise);
+        exercise = addTeamToExercise(team, exercise);
+
+        var participation = database.addTeamParticipationForExercise(exercise, team.getId());
+        var actualParticipation = request.get("/api/exercises/" + exercise.getId() + "/participation", HttpStatus.OK, StudentParticipation.class);
+        assertThat(actualParticipation).isEqualTo(participation);
+
+        participation = participationService.findOneByExerciseAndStudentLoginAnyStateWithEagerResultsElseThrow(exercise, student.getLogin());
+        assertThat(participation).isNotNull();
+        assertThat(participation.getId()).isEqualTo(participation.getId());
+    }
+
+    @Test
+    @WithMockUser(username = "student1", roles = "USER")
+    public void getParticipationByExerciseAndStudentLoginAnyStateForTeam() throws Exception {
+        var exercise = createTextExerciseForTeam();
+        var student = database.getUserByLogin("student1");
+        var team = createTeamForExercise(student, exercise);
+        exercise = addTeamToExercise(team, exercise);
+
+        var participation = database.addTeamParticipationForExercise(exercise, team.getId());
+        var actualParticipation = request.get("/api/exercises/" + exercise.getId() + "/participation", HttpStatus.OK, StudentParticipation.class);
+        assertThat(actualParticipation).isEqualTo(participation);
+
+        var participations = participationService.findOneByExerciseAndStudentLoginAnyState(exercise, student.getLogin());
+        assertThat(participations).isPresent();
+        assertThat(participations.get().getId()).isEqualTo(participation.getId());
+    }
```

Comment:
As it seems we have the same 7 lines (except for the GET of the students) over and over again, we could create a method something like 
`private Exercise initializeTeamExercise(User student) {
        var exercise = createTextExerciseForTeam();
        var team = createTeamForExercise(student, exercise);
        exercise = addTeamToExercise(team, exercise);
        var participation = database.addTeamParticipationForExercise(exercise, team.getId());
        var actualParticipation = request.get("/api/exercises/" + exercise.getId() + "/participation", HttpStatus.OK, StudentParticipation.class);
        assertThat(actualParticipation).isEqualTo(participation);`
and call this in every test after getting the student. The same like you did well with the other small private methods
PS: Markdown editor doesn't like my code linebreaks etc. but I think you know what I mean.
Rest looks very good 👍

Reply:
It looks like code duplication but the problem is that I use the `team` and `student` variables to assert things. So if I create a helper method which only returns an exercise, it would require more code to fetch the student etc.

Reply:
Yeah I did not include the student fetching in the helper, but I did not see that teams I indeed used again in a later assert, so your're right, looks like there is not much that can be done here, except for maybe also calling the teams method also outside the helper.